src/toolbars/AudioSetupToolBar.cpp

```diff
@@ -0,0 +1,840 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  AudioSetupToolBar.cpp
+
+*******************************************************************//*!
+
+\class AudioSetupToolBar
+\brief A toolbar to allow easier changing of input and output devices .
+
+*//*******************************************************************/
+
+
+
+#include "AudioSetupToolBar.h"
+#include "ToolManager.h"
+
+#include <thread>
+
+// For compilers that support precompilation, includes "wx/wx.h".
+#include <wx/wxprec.h>
+
+#include <wx/setup.h> // for wxUSE_* macros
+
+#ifndef WX_PRECOMP
+#include <wx/app.h>
+#include <wx/choice.h>
+#include <wx/event.h>
+#include <wx/intl.h>
+#include <wx/menu.h>
+#include <wx/menuitem.h>
+#include <wx/settings.h>
+#include <wx/sizer.h>
+#include <wx/statbmp.h>
+#include <wx/stattext.h>
+#include <wx/string.h>
+#include <wx/tooltip.h>
+#endif
+
+#include "../ActiveProject.h"
+#include "../TrackPanel.h"
+
+#include "AColor.h"
+#include "AllThemeResources.h"
+#include "AudioIOBase.h"
+#include "ImageManipulation.h"
+#include "DeviceToolBar.h"
+#include "../KeyboardCapture.h"
+#include "Prefs.h"
+#include "Project.h"
+#include "../ProjectWindows.h"
+#include "../ShuttleGui.h"
+#include "../widgets/Grabber.h"
+#include "DeviceManager.h"
+#include "../prefs/PrefsDialog.h"
+#include "../prefs/DevicePrefs.h"
+#include "../widgets/AButton.h"
+#include "../widgets/AudacityMessageBox.h"
+#include "../widgets/BasicMenu.h"
+#include "../widgets/Grabber.h"
+#include "../widgets/wxWidgetsWindowPlacement.h"
+
+#if wxUSE_ACCESSIBILITY
+#include "../widgets/WindowAccessible.h"
+#endif
```

Comment:
Add `#include <wx/log.h>`

You can eliminate
#include <wx/wxprec.h>
#include <wx/setup.h>
#include <wx/app.h>
#include <wx/choice.h>
#include <wx/event.h>
#include <wx/intl.h>
#include <wx/menuitem.h>
#include <wx/settings.h>
#include <wx/statbmp.h>
#include <wx/stattext.h>
#include <wx/string.h>
#include "../TrackPanel.h"
#include "AColor.h"
#include "ImageManipulation.h"
#include "Prefs.h"
#include "Project.h"
#include "../ShuttleGui.h"
#include "../widgets/Grabber.h"
#include "../widgets/AudacityMessageBox.h"
#include "../widgets/Grabber.h"
#include "../widgets/WindowAccessible.h"